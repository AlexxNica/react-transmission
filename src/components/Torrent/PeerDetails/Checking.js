import React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';

function Checking({ torrent, intl }) {
  const { recheckProgress } = torrent;

  return (
    <FormattedMessage
      id='torrent.full.checking'
      defaultMessage={`Verifying local data ({recheckProgress} tested)`}
      values={{
        recheckProgress: intl.formatNumber(recheckProgress, {style: 'percent'}),
      }}
    />
  );
}

export default injectIntl(Checking);
