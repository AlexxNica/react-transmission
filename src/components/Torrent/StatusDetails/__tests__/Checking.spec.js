import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Checking from '../Checking';

test('Checking', () => {
  const component = createComponentWithIntl(
    <Checking torrent={{recheckProgress: 0.5}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
