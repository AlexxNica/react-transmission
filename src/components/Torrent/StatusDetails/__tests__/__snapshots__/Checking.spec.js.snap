exports[`test Checking 1`] = `
<span>
  Verifying local data (50% tested)
</span>
`;
